slug.go

```diff
@@ -449,6 +432,56 @@ func Unpack(r io.Reader, dst string) error {
 	return nil
 }
 
+// Given a "root" directory, the path to a symlink within said root, and the
+// target of said symlink, checkSymlink checks that the target either falls
+// into root somewhere, or is explicitly allowed per the Packer's config.
+func (p *Packer) checkSymlink(root, path, target string) error {
+	// Get the absolute path to root.
+	absRoot, err := filepath.Abs(root)
+	if err != nil {
+		return fmt.Errorf("failed making path %q absolute: %w", root, err)
+	}
+
+	// Get the absolute path to the file path.
+	absPath := path
+	if !filepath.IsAbs(absPath) {
+		absPath = filepath.Join(absRoot, path)
+	}
+
+	// Get the absolute path of the symlink target.
+	var absTarget string
+	if filepath.IsAbs(target) {
+		absTarget = filepath.Clean(target)
+	} else {
+		absTarget = filepath.Join(filepath.Dir(absPath), target)
+	}
+
+	// Target falls within root.
+	if strings.HasPrefix(absTarget, absRoot) {
```

Comment:
This might be "belt plus suspenders", but Go has a function [`os.SameFile`](https://pkg.go.dev/os#SameFile), that compares whether two paths point at the same actual inode number (based on `stat`).  That would be one step beyond the string comparison.  However, there'd need to be extra logic if one or the other path doesn't exist.

If we used os.Samefile, there could be something like this pseudocode:

```
statTarget = os.Stat(strings.Left(absTarget, len(absRoot)))
statRoot = os.Stat(absRoot)
if os.SameFile(statRoot, statTarget) {
  // OK
} else {
  // the target is not within the root
}
```

Reply:
Yeah, we'd definitely need to augment to support the file-doesn't-exist case. We also can't really use this effectively during `Unpack()`, because we'd need to write the file before checking it with `os.SameFile` this way.

Reply:
OK sounds good.  I figured it might be a stretch but just wanted to mention it.